homeassistant/components/uvc/camera.py

```diff
@@ -35,26 +38,41 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up UVC integration."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_HOST: config["nvr"],
+                CONF_API_KEY: config["key"],
+                CONF_PASSWORD: config["password"],
+                CONF_PORT: config["port"],
+                CONF_SSL: config["ssl"],
+            },
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
     """Discover cameras on a Unifi NVR."""
-    addr = config[CONF_NVR]
-    key = config[CONF_KEY]
-    password = config[CONF_PASSWORD]
-    port = config[CONF_PORT]
-    ssl = config[CONF_SSL]
 
+    nvrconn: nvr.UVCRemote = hass.data[DOMAIN]["nvrconn"]
+    identifier: str = hass.data[DOMAIN]["camera_id_field"]
     try:
-        # Exceptions may be raised in all method calls to the nvr library.
-        nvrconn = nvr.UVCRemote(addr, port, key, ssl=ssl)
-        cameras = nvrconn.index()
-
-        identifier = "id" if nvrconn.server_version >= (3, 2, 0) else "uuid"
-        # Filter out airCam models, which are not supported in the latest
-        # version of UnifiVideo and which are EOL by Ubiquiti
+        cameras = await hass.async_add_executor_job(nvrconn.index)
         cameras = [
             camera
             for camera in cameras
-            if "airCam" not in nvrconn.get_camera(camera[identifier])["model"]
+            if "airCam"
+            not in (
+                await hass.async_add_executor_job(
+                    nvrconn.get_camera, camera[identifier]
+                )
+            )["model"]
         ]
     except nvr.NotAuthorized:
         _LOGGER.error("Authorization failure while connecting to NVR")
```

Comment:
Just `return` here after the log. Don't return `False`. A future improvement in another PR would be to add a reauth flow when authorization / authentication fails.